shared/types/Tournament.d.ts

```diff
@@ -1,27 +1,28 @@
-import User from './User';
+import User from "./User";
+import mongoose, { Schema, Document } from 'mongoose';
 
 export interface Team {
-  _id?: string;
   name: string;
   members: User[];
 }
 
 export interface Match {
-  id: string;
   rivals: {
     teamA: Team;
     teamB: Team;
   };
   score: {
-    teamA: number | null;
-    teamB: number | null;
+    teamA: number;
+    teamB: number;
```

Comment:
It is not correct. Score should be null when we didn't finish the game

Reply:
Got it, but how can we know if we finished game? It will be stored as a number in database. I suggest we set  "-1" value in these fields, before playing a match. You can't score -1 points. Another solution is to change database type from Number to Mixed or String

Reply:
@PrzemekRychter But we could to put here Number or Null. Is Mongodb support this?

Reply:
Keep in mind that we also need to account for score reporting here. Each team must report a score and only if they match the final score should be set and the match can be concluded.

Alternatively the final score can be forcefully set by the owner of the tournament when the sides do not agree on the score or they have not provided it at all.

Reply:
We can add 2 fields: `lastSetBy` which references the team and `confirmed`, e.g. member of Team A sets the score, then members of team B can confirm or reset that score. Confirmation would only be available if `lastSetBy` would referece the other team.

Reply:
I thought about it for a bit and I believe we should have a `score: { a: number, b: number }` for both `sideA` and `sideB` and also one for the whole match. This would allow every team to report a score for both sides and in the case of a disagreement the tournament owner would be able to compare them and set the final one to something else, keeping the history of the reported scores which might come in handy.

When you start implementing the score reporting we should have a short call about it.